modules/cli/src/main/java/org/apache/ignite/cli/builtins/cluster/ClusterApiClient.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.cli.builtins.cluster;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.util.List;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.ignite.cli.IgniteCliException;
+
+/**
+ * Client for Ignite Cluster management API.
+ */
+@Singleton
+public class ClusterApiClient {
+    private static final String CLUSTER_INIT_URL = "/management/v1/cluster/init/";
+
+    private final HttpClient httpClient;
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Inject
+    public ClusterApiClient(HttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    /**
+     * Sends 'cluster init' command to the specified node.
+     *
+     * @param nodeEndpoint host:port on which REST API is listening
+     * @param metastorageNodeIds consistent IDs of the nodes that will form metastorage RAFT group
+     * @param cmgNodeIds         consistent IDs of the nodes that will form Cluster Management RAFT Group; if empty,
+     *                           then nodes specified in metastorageNodeIds will form Cluster Management RAFT Group as well
+     * @param out                {@link PrintWriter} to which to report about the command outcome
+     */
+    public void init(String nodeEndpoint, List<String> metastorageNodeIds, List<String> cmgNodeIds, PrintWriter out) {
```

Comment:
Do you validate the `nodeEndpoint` format anywhere?

Reply:
No, the user will see an exception with the full URL containing any strange thing they will specify here. Do you believe we need to validatee it manually ourselves?

Reply:
Here is how it looks now:

```
java.lang.IllegalArgumentException: Illegal character in authority at index 7: http://some shit/management/v1/cluster/init/
```

Reply:
How do other input errors look like? If they look similarly, then I'm fine with the current approach